src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Read.HandlePropertyName.cs

```diff
@@ -45,6 +45,11 @@ public static partial class JsonSerializer
                     if (options.DictionaryKeyPolicy != null)
                     {
                         keyName = options.DictionaryKeyPolicy.ConvertName(keyName);
```

Comment:
Wait a second, why is the dictionary name policy used when deserializing? When was that decided?